design/behavior.system.md

```diff
@@ -0,0 +1,95 @@
+# Behavior System
+
+The behavior system is responsible for provinding the abstracted user interaction model for MRE Apps and how the user can interact with
+the various components of an MRE App.  It is based on the concept of high level behaviors that are applied to actors to make the actor
+"behave" like that particular interaction behavior.
+
+## Architecture
+
+The system is comprised of several layers and components that make up the system.
+
+### SDK Components and Layers
+
+- Behavior
+    - Actions
+    - Action Handlers
+    - Action Data
+
+### Runtime Components and Layers
+
+- Behavior Context
+    - Action Handlers
+    - Action Data
+- Behavior Plugin Interface
+- Behavior Factory Plugin Interface
+
+## Network
+
+- SetBehavior Payload
+- PerformAction Payload
+
+## Sync layer considerations
+
+We sync behaviors from all clients in the server.  All filtering is handled in the runtime.
+
+## Client implementation
+
+Client runtime is resposible for getting the concrete host app implementation of the behaviors through the host app implementation
+of the `IBehaviorFactory`, and supplying the necesssary context for the host app to raise input events on.  The permission system is
+responsible for enabling or disabling the behavior system per user, or globally on the client if there is no user interacting with the
+app at all for that client.
```

Comment:
What does it mean for there to be usage but no user?

Reply:
Not sure what you mean?  As in have user interaction but not a user?

---
In reply to: [448607042](https://github.com/microsoft/mixed-reality-extension-sdk/pull/649#discussion_r448607042) [](ancestors = 448607042)

Reply:
I think I figured this statement out, but to clarify my original question: what would it mean for the permission system to enable interaction globally if there are no users? I think the answer is it wouldn't, and this comment refers to disabling only.

---
In reply to: [448618887](https://github.com/microsoft/mixed-reality-extension-sdk/pull/649#discussion_r448618887) [](ancestors = 448618887,448607042)

Reply:
Right.  It wouldn't be enabled globally for no users.